### PR TITLE
[Keyboard] Add GTM Pad macropad

### PR DESCRIPTION
**Pull request by jrfhoutx**
## Description

Adding support for GTM Pad macropad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sck/gtm/readme.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Grand Theft Macro Pad
+
+![Grand Theft Macropad](https://i.imgur.com/ON7m7RI.jpg)
+
+A 2x5 macropad/numpad with a rotary encoder, cloned from the 2Key2Crawl macropad that was exclusively available at Austin Keycrawl 2018 (12-01-2018).
+
+Keyboard Maintainer: jrfhoutx
```
Comment (fauxpark) [inline keyboards/sck/gtm/readme.md]:
```suggestion
Keyboard Maintainer: jrfhoutx  
```

File: keyboards/sck/gtm/readme.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+Keyboard Designer: [jrfhoutx](https://github.com/jrfhoutx)  
+Hardware Supported: GTM Pad
```
Comment (fauxpark) [inline keyboards/sck/gtm/readme.md]:
```suggestion
Hardware Supported: GTM Pad  
```

File: keyboards/sck/gtm/rules.mk [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# MCU name
+MCU = atmega32u2
+
+
```
Comment (fauxpark) [inline keyboards/sck/gtm/rules.mk]:
```suggestion
```

File: keyboards/sck/gtm/rules.mk [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
+BOOTLOADER = atmel-dfu
+
```
Comment (fauxpark) [inline keyboards/sck/gtm/rules.mk]:
```suggestion
```

File: keyboards/sck/gtm/rules.mk [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (fauxpark) [inline keyboards/sck/gtm/rules.mk]:
```suggestion
```

File: keyboards/sck/gtm/rules.mk [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration(+1000)
```
Comment (fauxpark) [inline keyboards/sck/gtm/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
```

File: keyboards/sck/gtm/rules.mk [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+MOUSEKEY_ENABLE = no	# Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/sck/gtm/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no	# Mouse keys
```

File: keyboards/sck/gtm/rules.mk [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+EXTRAKEY_ENABLE = no	# Audio control and System control(+450)
```
Comment (fauxpark) [inline keyboards/sck/gtm/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = no	# Audio control and System control
```

File: keyboards/sck/gtm/rules.mk [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+CONSOLE_ENABLE = yes	# Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/sck/gtm/rules.mk]:
```suggestion
CONSOLE_ENABLE = yes	# Console for debug
```

File: keyboards/sck/gtm/rules.mk [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+MCU = atmega32u2
+
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
```
Comment (fauxpark) [inline keyboards/sck/gtm/rules.mk]:
```suggestion
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

File: keyboards/sck/gtm/config.h [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+#define PRODUCT         GTM Pad
+#define DESCRIPTION     Grand Theft Macro Pad
+
+/* key matrix size */
+#define MATRIX_ROWS 3
+#define MATRIX_COLS 6
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS { C4, C5, D1 }
+#define MATRIX_COL_PINS { B4, B5, B6, B7, C7, D0 }
```
Comment (noroadsleft) [inline keyboards/sck/gtm/config.h]: According to

https://github.com/qmk/qmk_firmware/blob/a5475a826d5ee162a5e71b7c16e5daf7ab6de7ff/keyboards/sck/gtm/gtm.h#L5-L12

the matrix here is only 3x5. Shouldn't `MATRIX_COLS` be `5` and `MATRIX_COL_PINS` adjusted accordingly?

Comment (jrfhoutx) [reply keyboards/sck/gtm/config.h]: the push button function of the rotary encoder is on it's own row and column, so it should be 3x6, not 3x5. I've now adjusted the layout to correctly account for that.

File: keyboards/sck/gtm/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+void matrix_init_user(void) {
+  debug_config.matrix = 1;
+  debug_config.keyboard = 1;
+  debug_config.enable = 1;
+}
+
+
+
+void encoder_update_user(int8_t index, bool clockwise) {
+  if (index == 0) {
```
Comment (drashna) [inline keyboards/sck/gtm/keymaps/default/keymap.c]: If you only have the one, you don't really need to have the index.  

However, there is also no harm in having it, either.

Comment (jrfhoutx) [reply keyboards/sck/gtm/keymaps/default/keymap.c]: so then lines 19-27 can be removed?

File: keyboards/sck/gtm/config.h (hunk 11 above)
Comment (noroadsleft) [reply keyboards/sck/gtm/config.h]: Resolved in https://github.com/qmk/qmk_firmware/pull/7123/commits/0a656ce9c9b78cdd03d3e0c1e635032b37c84609.

File: keyboards/sck/gtm/keymaps/default/keymap.c (hunk 12 above)
Comment (fauxpark) [reply keyboards/sck/gtm/keymaps/default/keymap.c]: Just the enclosing `if (index == 0) { ... }`.

Comment (jrfhoutx) [reply keyboards/sck/gtm/keymaps/default/keymap.c]: ok I think I understand. that should be correct now. if not please let me know.

**Comment by noroadsleft** [comment]
Looks good, thanks!

**Pull request merged**